### PR TITLE
docs: Add docs for LambdaDSL, with minor style fixes

### PR DESCRIPTION
**Pull request by joklek**
Add docs for LambdaDSL, mainly copied from the old method javadocs. Automatic fixes for styling, renamed a few parameters, to better reflect their use

EDIT: Just noticed this https://github.com/pact-foundation/pact-jvm/releases/tag/4_2_0-beta_3 and it seems like the classes I worked on will be replaced with Kotlin :( At least they have their documentation added

### CODE REVIEW

File: consumer/java8/src/main/java/io/pactfoundation/consumer/dsl/LambdaDslJsonArray.java [hunk 1]
```diff
@@ -448,81 +568,100 @@ public LambdaDslJsonArray eachArrayWithMinLike(int numberExamples, Integer size,
...
+     *
+     * @param expression Datetime expression
+     * @param format     Datetime format to use for values
+     */
+    public LambdaDslJsonArray datetimeExpression(String expression, String format) {
+        pactArray.datetimeExpression(expression, format);
+        return this;
+    }
 
     public DslPart build() {
```
Comment (joklek) [inline consumer/java8/src/main/java/io/pactfoundation/consumer/dsl/LambdaDslJsonArray.java:664]: Moved build() down, to be at the bottom of the file

File: consumer/java8/src/main/java/io/pactfoundation/consumer/dsl/LambdaDsl.java [hunk 2]
```diff
@@ -13,13 +14,22 @@
  */
 public class LambdaDsl {
 
+    private LambdaDsl() {
```
Comment (joklek) [inline consumer/java8/src/main/java/io/pactfoundation/consumer/dsl/LambdaDsl.java:17]: Make implicit public no-arg constructor private just to be as tidy as possible

**Comment by uglyog** [comment]
@joklek these classes haven't been changed yet, I think it is still valuable to merge this?

**Comment by joklek** [comment]
Ah, my internal pessimist won without any response :D Thanks for your input, I reopened the PR

**Pull request merged**